plugins/catalog/src/alpha/components/EntityTabs/EntityTabsGroup.tsx

```diff
@@ -138,22 +142,41 @@ const styles = (theme: Theme) =>
 
 type EntityTabsGroupItem = {
   id: string;
-  index: number;
   label: string;
   path: string;
-  group: string;
+  icon?: string | ReactElement;
 };
 
 type EntityTabsGroupProps = TabProps & {
   classes?: Partial<ReturnType<typeof styles>>;
   indicator?: ReactNode;
-  highlightedButton?: number;
+  highlightedButton?: string;
   items: EntityTabsGroupItem[];
-  onSelectTab: MouseEventHandler<HTMLAnchorElement>;
+  onSelectTab?: MouseEventHandler<HTMLAnchorElement>;
+  showIcons?: boolean;
 };
 
+function resolveIcon(
+  icon: string | ReactElement | undefined,
+  iconsApi: IconsApi,
+  showIcons: boolean,
+) {
+  if (!showIcons) {
+    return undefined;
+  }
+  const itemIcon = icon;
```

Comment:
you don't need this extra variable do you? the type checker should be fine with narrowing to just `string` inside the `if`, directly for the `icon` parameter